backend/projects/signals.py

```diff
@@ -1,9 +1,18 @@
-from django.db.models.signals import post_save, post_delete
+from django.db.models.signals import pre_save, post_save, post_delete
 from django.dispatch import receiver
+from django.conf import settings
 
 from projects.models import Project
 from projects.rdf_models import RDFProject
 
+@receiver(pre_save, sender=Project)
+def set_project_uri(sender, instance: Project, **kwargs):
+    '''
+    Set project URI if it is empty.
+    '''
+    if not instance.uri:
+        instance.uri = settings.RDF_NAMESPACE_ROOT + 'projects/' + instance.name
```

Comment:
Here's a second place where you define the URI of a project. This one is probably better. Regardless, you should factor out the logic to a single source of truth.